lib/trie/recorder.go

```diff
@@ -0,0 +1,40 @@
+package trie
+
+// NodeRecord represets a record of a visited node
+type NodeRecord struct {
```

Comment:
don't think the types/funcs in here need to be exported either

Reply:
Updated!